ppl/src/main/java/com/amazon/opendistroforelasticsearch/sql/ppl/parser/AstBuilder.java

```diff
@@ -197,6 +203,62 @@ public UnresolvedPlan visitEvalCommand(EvalCommandContext ctx) {
     );
   }
 
+  /**
+   * Rare command.
+   */
+  @Override
+  public UnresolvedPlan visitRareCommand(RareCommandContext ctx) {
+    List<UnresolvedExpression> groupList = ctx.byClause() == null ? Collections.emptyList() :
+        ctx.byClause()
+            .fieldList()
+            .fieldExpression()
+            .stream()
+            .map(this::visitExpression)
+            .collect(Collectors.toList());
+    return new RareTopN(
+
+        /**
+         * Setting rareTopFlag to FALSE will return list of rare values
+         */
+        Boolean.FALSE,
+        ArgumentFactory.getArgumentList(ctx),
+        ctx.fieldList()
+            .fieldExpression()
+            .stream()
+            .map(field -> (Field) visitExpression(field))
+            .collect(Collectors.toList()),
+        groupList
+    );
+  }
+
+  /**
+   * Top command.
+   */
+  @Override
+  public UnresolvedPlan visitTopCommand(TopCommandContext ctx) {
```

Comment:
Code is the same for these two methods, except differing by the flag. Can they be combined into a helper? If the two context classes have the same methods...

Reply:
Added helper methods for getting a list of groups & fields. Can't add other arguments in helper since that requires respective command context object